components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -0,0 +1,1955 @@
+                // We support only 'REPLACE' patch operation.
+                if (operation == Patch.OperationEnum.REPLACE) {
+                    String value = null;
+                    if (patch.getValue() instanceof String) {
+                        value = (String) patch.getValue();
+                    }
+                    switch (path) {
+                        case Constants.NAME_PATH:
+                            idpToUpdate.setIdentityProviderName(value);
+                            break;
+                        case Constants.DESCRIPTION_PATH:
+                            idpToUpdate.setIdentityProviderDescription(value);
+                            break;
+                        case Constants.IMAGE_PATH:
+                            idpToUpdate.setImageUrl(value);
+                            break;
+                        case Constants.IS_PRIMARY_PATH:
+                            idpToUpdate.setPrimary(Boolean.parseBoolean(value));
+                            break;
+                        case Constants.IS_ENABLED_PATH:
+                            idpToUpdate.setEnable(Boolean.parseBoolean(value));
+                            break;
+                        case Constants.IS_FEDERATION_HUB_PATH:
+                            idpToUpdate.setFederationHub(Boolean.parseBoolean(value));
+                            break;
+                        case Constants.HOME_REALM_PATH:
+                            idpToUpdate.setHomeRealmId(value);
+                            break;
+                        default:
+                            throw handleException(Response.Status.BAD_REQUEST, Constants.ErrorMessage
+                                    .ERROR_CODE_INVALID_INPUT, null);
+                    }
+                } else {
+                    // Throw an error if any other patch operations are sent in the request.
+                    throw handleException(Response.Status.BAD_REQUEST, Constants.ErrorMessage
+                            .ERROR_CODE_INVALID_INPUT, null);
+                }
+            }
+            IdentityProvider updatedIdP = IdentityProviderServiceHolder.getIdentityProviderManager()
+                    .updateIdPByResourceId(identityProviderId, idpToUpdate,
+                            ContextLoader.getTenantDomainFromContext());
+            return createIDPResponse(updatedIdP);
+
+        } catch (IdentityProviderManagementException e) {
+            throw handleIdPException(e, Constants.ErrorMessage.ERROR_CODE_ERROR_RETRIEVING_IDP, identityProviderId);
+        }
+    }
+
+    /**
+     * Delete an Identity Provider.
+     *
+     * @param identityProviderId Identity Provider resource ID.
+     * @param force              when force is set to true, IDP will be forcefully deleted even-though it is being
+     *                           referred by service providers.
+     */
+    public void deleteIDP(String identityProviderId, boolean force) {
```

Comment:
since the whole implementation of the method differs from the `boolean force` parameter, better to separate them into 2 separate methods. `deleteIDP` and `forceDeleteIDP`